rcav2/agent/react.py

```diff
@@ -55,7 +74,76 @@ async def search_errors(regex: str) -> list[rcav2.models.errors.LogFile]:
                     break
         return logfiles
 
-    return dspy.ReAct(RCAAccelerator, tools=[read_errors, search_errors])
+    async def search_jira_issues(
+        query: str, max_results: int | None = 50
+    ) -> list[dict[str, str | None]]:
+        """Searches jira issues using JQL (Jira query language).
+        Returns list of issues with key, url, summary, status, and description.
+        The 'url' field contains the full link to the JIRA ticket.
+        Returns 50 results by default, for more results set max_results.
+        Use the 'key' field from results with get_jira_issue for more details.
+        If JIRA_RCA_PROJECT is configured, automatically filters to that project.
+
+        JQL Query Syntax - IMPORTANT:
+        - Text search: text ~ "error message" (quotes required for phrases)
+        - Summary search: summary ~ "keyword"
+        - Description search: description ~ "error text"
+        - Multiple terms: summary ~ "cert-manager" AND text ~ "timeout"
+        - OR condition: summary ~ "error" OR description ~ "failure"
+
+        Valid operators: ~ (contains), !~, =, !=, IN, NOT IN
+        Always use ~ for text searches with quoted strings."""
+        if not env.jira_client:
+            await worker.emit(
+                "JIRA client not available. Set JIRA_URL and JIRA_API_KEY", "error"
+            )
+            return []
+
+        # Add project filter if configured
+        final_query = query
+        if env.jira_rca_project:
+            # If query doesn't already contain "project =", add it
+            if "project" not in query.lower():
+                # Support multiple projects (comma-separated)
+                projects = [p.strip() for p in env.jira_rca_project.split(",")]
+                if len(projects) == 1:
+                    final_query = f"project = {projects[0]} AND ({query})"
+                else:
+                    project_list = ", ".join(projects)
+                    final_query = f"project IN ({project_list}) AND ({query})"
+
+        await worker.emit(
+            f"Searching issues with query: {final_query}, max_results: {max_results}",
+            "progress",
+        )
+        try:
+            result = env.jira_client.search_issues(final_query, maxResults=max_results)
+            # Convert Issue objects to serializable dicts
+            jira_base_url = env.jira_client._options["server"]
+            result_list = []
+            for issue in result if result else []:
+                issue_dict = {
+                    "key": issue.key,
+                    "url": f"{jira_base_url}/browse/{issue.key}",
+                    "summary": getattr(issue.fields, "summary", ""),
```

Comment:
According to the signature, this should be `getattr(issue.fields, "summary", None)`